src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,141 @@
+     */
+    static Command parse(String fullCommand) throws DukeException {
+        String[] commandInputs = fullCommand.trim().split(" ", 2);
+
+        if (commandInputs.length == 0) {
+            throw new DukeException("Something went wrong when parsing your inputs!");
+        }
+
+        String command = commandInputs[0].trim().toLowerCase();
+
+        String commandDetails;
+
+        switch (command) {
+        case "list":
+            return new ListCommand();
+
+        case "find":
+            // "find" needs to be accompanied by a keyword
+            if (commandInputs.length < 2) {
+                throw new MissingKeywordException("I'm not sure what tasks to search for...");
+            }
+
+            commandDetails = commandInputs[1].trim();
+
+            return new FindCommand(commandDetails);
+
+        case "done":
+            // "done" needs to be accompanied by task ID
+            if (commandInputs.length < 2) {
+                throw new MissingTaskIdException(FAILED_TO_MARK_TASK_AS_COMPLETE);
+            }
+
+            commandDetails = commandInputs[1].trim();
+
+            try {
+                int taskId = Integer.parseInt(commandDetails);
+                return new DoneCommand(taskId);
+            } catch (NumberFormatException e) {
+                throw new InvalidTaskIdException(FAILED_TO_MARK_TASK_AS_COMPLETE);
+            }
+
+        case "delete":
+            // "delete" needs to be accompanied by task ID
+            if (commandInputs.length < 2) {
+                throw new MissingTaskIdException(FAILED_TO_DELETE_TASK);
+            }
+
+            commandDetails = commandInputs[1].trim();
+
+            try {
+                int taskId = Integer.parseInt(commandDetails);
+                return new DeleteCommand(taskId);
+            } catch (NumberFormatException e) {
+                throw new InvalidTaskIdException(FAILED_TO_DELETE_TASK);
+            }
+
+        case "todo":
+        case "deadline":
+        case "event":
+            // "todo", "deadline", "event" needs to be accompanied with details on the task to be
+            // created
+            if (commandInputs.length < 2) {
+                throw new MissingTaskDetailsException(FAILED_TO_CREATE_TASK);
+            }
+
+            commandDetails = commandInputs[1].trim();
+
+            if (command.equals("todo")) {
+                return new AddCommand(new Todo(commandDetails));
+            } else if (command.equals("deadline")) {
+                String[] deadlineDetails = commandDetails.split("/by", 2);
+
+                if (deadlineDetails.length < 2) {
+                    throw new MissingTaskDetailsException(FAILED_TO_CREATE_DEADLINE_TASK,
+                            "No deadline was specified!");
+                }
+
+                String description = deadlineDetails[0].trim();
+                String by = deadlineDetails[1].trim();
+
+                return new AddCommand(new Deadline(description, by));
```

Comment:
Maybe you could separate out logic such as this into its own method? This would help make the method shorter.